### DISCUSSION TITLE
Can we replace WinUAE with newer versions?

### DISCUSSION THREAD

**Discussion by BODYPRINT** [prb28/vscode-amiga-assembly#278] [Q&A] [answered]
I tried all the latest versions from 4.9.1 to 5.0.0 and the "WinUAE Launch" works fine, but there's no Debug support. "WinUAE Debug" just runs like a standard Launch version.
Is there a trick to using a newer version of WinUAE than 4.9.0 with Debug support?
Cheers!

**Comment by BODYPRINT** [comment, marked as answer]
Sorry, I just found the answer in the UAE-DAP Project.
Is there any chance of patching a newer version of WinUAE?
Thanks again for this amazing project.

**Reply by prb28** [reply]
Thanks.

As you have found, a custom build of winuae is needed.
This essential part of the extension is now in the hands of @grahambates who is the developer of uae-dap. So I'll let him answer about the new versions of the emulator.

**Reply by grahambates** [reply]
Yeah absolutely. uae-dap currently still uses the https://github.com/prb28/WinUAE fork, which is now quite far behind. Bartmans original fork has been kept relatively up to date though, so it shouldn't be a big deal to resync the changes. I'll give it a go.

**Reply by BODYPRINT** [reply]
Thanks guys :)

**Comment by BODYPRINT** [comment]
Hi` @grahambates and @prb28 
I updated my winuae.exe to Graham's new UAE-DAP version 5.1.0. The Launch versions work, but my Debug versions don't seem to be connecting to VS-Code. It just hangs and no debug options show in VS-Code.
Do I need to make any additional changes?

I'm using the same .uae files as I used for the previous verion (4.10.1) which still works after the latest updates to VS-Code and 1.8.7 of the extension.

```
            "type": "amiga-assembly",
            "request": "launch",
            "name": "WinUAE Debug A1200 FPU",
            "stopOnEntry": true,
            "program": "${workspaceFolder}/uae/dh0/game",
            "remoteProgram": "SYS:game",
            "emulatorType": "winuae",
            "emulatorArgs": [
                "-f",
                "MyA1200FPU.uae",
                "-s",
                "filesystem=rw,dh0:${workspaceFolder}/uae/dh0",
            ],
            "preLaunchTask": "amigaassembly: build A1200 FPU"
        },

        {
            "type": "amiga-assembly",
            "request": "launch",
            "name": "WinUAE Debug A500",
            "stopOnEntry": true,
            "program": "${workspaceFolder}/uae/dh0/game",
            "remoteProgram": "SYS:game",
            "emulatorType": "winuae",
            "emulatorArgs": [
                "-f",
                "MyA500.uae",
                "-s",
                "filesystem=rw,dh0:${workspaceFolder}/uae/dh0",
            ],
            "preLaunchTask": "amigaassembly: build A500"
        },

```

Thanks again for the amazing work.
